CVAE_testbed/datasets/quilt_aics_features.py

```diff
@@ -0,0 +1,239 @@
+import torch
+from quilt3 import Package
+from collections import defaultdict
+from torch.utils.data import Dataset
+import pandas as pd
+import os, json
+from featuredb import FeatureDatabase
+from sklearn import preprocessing
+import numpy as np
+from scipy import stats
+from scipy.stats import zscore
+import sklearn
+
+
+class QuiltAicsFeatures(Dataset):
+    def __init__(self, num_batches, BATCH_SIZE, model_kwargs, shuffle=True, corr=False, train=True, mask=False):
+        """
+        Args: 
+            num_batches: Number of batches of synthetic data
+            BATCH_SIZE: batchsize of synthetic data
+            model_kwargs: dictionary containing "x_dim" which indicates input data size
+            shuffle:  True sets condition vector in input data to 0 for all possible permutations
+            corr: True sets dependent input dimensions via a correlation matrix 
+        """
+        self.num_batches = num_batches
+        self.BATCH_SIZE = BATCH_SIZE
+        self.corr = corr
+        self.shuffle = shuffle
+        self.model_kwargs = model_kwargs
+        self.train = train
+
+        Batches_X_train, Batches_C_train, Batches_conds_train = torch.empty([0]), torch.empty([0]), torch.empty([0])
+        Batches_X_test, Batches_C_test, Batches_conds_test = torch.empty([0]), torch.empty([0]), torch.empty([0])
+
+        ds = Package.browse("aics/pipeline_integrated_single_cell", "s3://allencell")
+
+        # Specify path to pre downloaded quilt json files
+        try:
+            path_to_json = model_kwargs['json_quilt_path']
+        except:
+            path_to_json = "/home/ritvik.vasan/test/"
+        json_files = [pos_json for pos_json in os.listdir(path_to_json) if pos_json.endswith('.json')]
+
+        meta_to_file_name = []
+        for f in ds["cell_features"]:
+            meta_to_file_name.append({"filename": f, **ds["cell_features"][f].meta})
+        metas = pd.DataFrame(meta_to_file_name)
+
+        # Specify path to config file for FeatureDatabase
+        try:
+            db = FeatureDatabase(model_kwargs['config_path'])
```

Comment:
Note to me: This makes me think I should really combine a lot of our various feature / data sets together into one big one, or at least, combine all the various integrated cell feature / data sets...